core/domain/learner_progress_services.py

```diff
@@ -420,3 +470,46 @@ def get_all_incomplete_collection_ids(user_id):
         return incomplete_activities.collection_ids
     else:
         return []
+
+
+def remove_none_entities_from_list(
+        user_id, entity_summary_list, entity_ids, section, sub_section):
+    """Removes all the None entries from a list and returns it.
+
+    Args:
+        user_id: str. The id of the user.
+        entity_summary_list: list(Summary|None). A list of the summary domain
+            objects.
+        entity_ids: list(str). The ids of the entities.
+        section: str. To which section does the entities belong to. Can be
+            'complete' or 'incomplete'.
+        sub_section: str. To which the sub-section do the entities belong to.
+            Can be 'explorations' or 'collections'.
+
+    Returns:
+        list(Summary). A list of summary domain objects.
+        int. The number of none entities in the list.
+    """
+    number_removed = 0
+    index = 0
+    filtered_entity_summary_list = []
+    for summary in entity_summary_list:
+        if summary is None:
+            number_removed = number_removed + 1
+            if section == 'complete':
+                if sub_section == 'explorations':
```

Comment:
Just as a side note, these should really be named constants, not "magic strings". This might be obviated by the proposed refactor (see other comments) but I wanted to point it out for future reference.